File: lib-cli/tmp/server-ts/src/producer/implementation/events/connected.ts
import { Producer, Identification, Filter } from "./index";
import { emit } from "../../events/connected";

export function handler<C>(
	consumer: Identification,
	filter: Filter,
	context: C,
	producer: Producer<C>
): Promise<void> {
	return emit(consumer, filter, context, producer);
}

File: experimental/workflow_mt/producer/rust/src/producer/implementation/emitters/connected.rs
use super::{events, hub, identification, producer::Control, Context, EmitterError};
use clibri::server;
pub async fn emit<E: server::Error, C: server::Control<E> + Send + Clone>(
    identification: &identification::Identification,
    filter: hub::filter::Filter,
    context: &Context,
    control: &Control<E, C>,
) -> Result<(), EmitterError> {
    events::connected::emit::<E, C>(identification, filter, context, control)
        .await
        .map_err(EmitterError::Emitting)
}
